jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/AbstractEJBServicesTest.java

```diff
@@ -95,9 +96,20 @@ public boolean accept(File dir, String name) {
             }
         }
     }
+ 
+    protected static Throwable catchRootCause(ThrowingCallable shouldRaiseThrowable) {
+        Throwable result = Assertions.catchThrowable(shouldRaiseThrowable);
+        Throwable cause = null; 
+
+        while(null != (cause = result.getCause())  && (result != cause) ) {
```

Comment:
Interesting, so down the stacktrace it can happen that `result == clause`?

Reply:
It shouldn't,  but it covers possible mistakes when somebody sets the cause of the exceptions to itself (that would cause an infinite loop). I haven't seen it here but it does not hurt

Reply:
@elguardian yeah, definitely